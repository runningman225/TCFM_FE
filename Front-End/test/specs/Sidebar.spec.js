
import { createLocalVue, shallowMount } from '@vue/test-utils'
import Sidebar from '@/Components/Sidebar'
import Vuex from 'vuex'
import VueRouter from 'vue-router';


let wrapper
let store
let actions
let mutations
let state
let router = new VueRouter()
const localVue = createLocalVue()

localVue.use(VueRouter)
localVue.use(Vuex)

beforeEach(()  => {
    wrapper = shallowMount(Sidebar, {
        propsData: {},
        mocks: { $t: () => {} },
        stubs: {},
        store,
        localVue,   
        router, 
    });
})
afterEach(() => {
    wrapper.destroy()
})

describe('Components/Sidebar.vue',()=>{
    test('renders correctly', () => {
        expect(wrapper.element).toMatchSnapshot()
        expect(wrapper).toBeTruthy()
      })
      
    test('test function changeView', ()=>{
        wrapper.vm.$router.push = jest.fn()
        wrapper.vm.changeView()
        expect(wrapper.vm.$router.push).toBeCalled()
    })
    test('test function backtoLogin',()=>{
        wrapper.vm.$router.push = jest.fn()
        wrapper.vm.backToLogin()
        expect(wrapper.vm.$router.push).toBeCalled()
    })
    test('dashboard menu active',()=>{
        wrapper.setData({

        })
        wrapper.vm.changeView = jest.fn(()=>{
            wrapper.vm.isDashboard = true
        })
        const menu = wrapper.find('.dashBoard .menuWrapper')
        wrapper.vm.changeView('/dashboard')
        expect(wrapper.vm.changeView).toBeCalled()
        expect(wrapper.vm.isDashboard).toBe(true)
    })
    test('role admin\'s menu',()=>{
        expect(wrapper.text()).not.toContain('Management')

        wrapper.setData({
            isAdmin : true
        })
        expect(wrapper.text()).toContain('Management')
    })


})