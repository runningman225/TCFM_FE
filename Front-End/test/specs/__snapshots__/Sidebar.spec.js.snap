// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Sidebar.vue renders correctly 1`] = `
<div
  class="sidebarComponent"
>
  <div>
    <div
      class="logoBlibli"
    >
      <img
        alt="Logo Blibli"
        class="logoBlibli"
        src="../assets/logo-blibli.png"
      />
    </div>
     
    <div
      class="menuContainer"
    >
      <div
        class="menuParent dashBoard"
      >
        <div
          class="littleBlue"
          style="display: none;"
        />
         
        <div
          class="menuWrapper"
        >
          <img
            alt="Dashboard"
            class="menuIcon"
            src="../assets/dashboard-icon.png"
          />
          
          Dashboard
        
        </div>
      </div>
       
      <div
        class="menuParent"
      >
        <div
          class="menuWrapper"
        >
          <img
            alt="Dashboard"
            class="menuIcon"
            src="../assets/detail-icon.png"
          />
          
          Group Detail
        
        </div>
         
        <div
          class="spacer"
        />
         
        <div
          class="menuChild"
        >
          <div
            class=""
          />
           
          <div
            class="menuChildText"
          >
            Overview
          </div>
           
          <div />
        </div>
         
        <div
          class="menuChild"
        >
          <div
            class=""
          />
           
          <div
            class="menuChildText"
          >
            Expenses
          </div>
           
          <div />
        </div>
         
        <div
          class="menuChild"
        >
          <div
            class=""
          />
           
          <div
            class="menuChildText"
          >
            Members
          </div>
           
          <div />
        </div>
      </div>
       
      <div
        class="menuParent"
      >
        <div
          class="littleBlue"
          style="display: none;"
        />
         
        <div
          class="menuWrapper"
        >
          <img
            alt="Dashboard"
            class="menuIcon"
            src="../assets/invoice-icon.png"
          />
          
          Payment
        
        </div>
         
        <div
          class="spacer"
        />
         
        <div
          class="menuChild"
        >
          <div
            class=""
          />
           
          <div
            class="menuChildText"
          >
            History
          </div>
           
          <div />
        </div>
         
        <div
          class="menuChild"
        >
          <div
            class=""
          />
           
          <div
            class="menuChildText"
          >
            Manage
          </div>
           
          <div />
        </div>
      </div>
       
      <!---->
    </div>
  </div>
   
  <div
    class="logoutButton"
  >
    
    Logout
  
  </div>
</div>
`;
